homeassistant/components/ovo_energy/sensor.py

```diff
@@ -32,199 +52,148 @@ async def async_setup_entry(
-            "kWh",
-        )
-
-    @property
-    def native_value(self) -> str:
-        """Return the state of the sensor."""
-        usage: OVODailyUsage = self.coordinator.data
-        if usage is None or not usage.gas:
-            return None
-        return usage.gas[-1].consumption
-
-    @property
-    def extra_state_attributes(self) -> object:
-        """Return the attributes of the sensor."""
-        usage: OVODailyUsage = self.coordinator.data
-        if usage is None or not usage.gas:
-            return None
-        return {
-            "start_time": usage.gas[-1].interval.start,
-            "end_time": usage.gas[-1].interval.end,
-        }
-
-
-class OVOEnergyLastElectricityCost(OVOEnergySensor):
-    """Defines a OVO Energy last cost sensor."""
-
-    def __init__(
-        self, coordinator: DataUpdateCoordinator, client: OVOEnergy, currency: str
     ) -> None:
-        """Initialize OVO Energy sensor."""
-        super().__init__(
-            coordinator,
-            client,
-            f"{DOMAIN}_{client.account_id}_last_electricity_cost",
-            "OVO Last Electricity Cost",
-            "mdi:cash-multiple",
-            DEVICE_CLASS_MONETARY,
-            currency,
-        )
+        """Initialize."""
+        super().__init__(coordinator, client)
+        self.entity_description = description
 
     @property
-    def native_value(self) -> str:
-        """Return the state of the sensor."""
-        usage: OVODailyUsage = self.coordinator.data
-        if usage is None or not usage.electricity:
-            return None
-        return usage.electricity[-1].cost.amount
-
-    @property
-    def extra_state_attributes(self) -> object:
-        """Return the attributes of the sensor."""
-        usage: OVODailyUsage = self.coordinator.data
-        if usage is None or not usage.electricity:
-            return None
-        return {
-            "start_time": usage.electricity[-1].interval.start,
-            "end_time": usage.electricity[-1].interval.end,
-        }
-
-
-class OVOEnergyLastGasCost(OVOEnergySensor):
-    """Defines a OVO Energy last cost sensor."""
-
-    def __init__(
-        self, coordinator: DataUpdateCoordinator, client: OVOEnergy, currency: str
-    ) -> None:
-        """Initialize OVO Energy sensor."""
-        super().__init__(
-            coordinator,
-            client,
-            f"{DOMAIN}_{client.account_id}_last_gas_cost",
-            "OVO Last Gas Cost",
-            "mdi:cash-multiple",
-            DEVICE_CLASS_MONETARY,
-            currency,
-        )
-
-    @property
-    def native_value(self) -> str:
-        """Return the state of the sensor."""
-        usage: OVODailyUsage = self.coordinator.data
-        if usage is None or not usage.gas:
-            return None
-        return usage.gas[-1].cost.amount
+    def unique_id(self) -> str:
+        """Return the unique ID for this entity."""
+        return self.entity_description.key
```

Comment:
This can be defined as attribute in `__init__`: `_attr_unique_id`. Together with `name_prefix`, this will probably look something like this:
```py
self._attr_unique_id = f"{name_prefix}_{description}`
```